numpy/f2py/rules.py

```diff
@@ -124,6 +124,10 @@
 #define PY_SSIZE_T_CLEAN
 #endif /* PY_SSIZE_T_CLEAN */
 
+/* Unconditionally included */
+#include <Python.h>
+#include <numpy/npy_os.h>
```

Comment:
Technically `npy_os.h` should only be included if `long_long` or `unsigned_long_long` or threading is required.

Reply:
Another option which would also provide quite a bit of clean up is to use `#include <numpy/npy_common.h>` and remove the `typedef`s used by F2PY for say, complex numbers.

Reply:
In keeping with the original "minor cleanup" spirit of this PR, we will only use `npy_os.h` here and defer the larger cleanup w.r.t. `npy_common.h` to #21161.